cvat/apps/engine/log.py

```diff
@@ -49,108 +19,8 @@ def get_logger(logger_name, log_file):
     return logger
 
-def _close_logger(logger: logging.Logger):
-    for handler in logger.handlers:
-        handler.close()
-
-class LogManager:
-    def close(self):
-        raise NotImplementedError
-
-class IndexedLogManager(LogManager):
-    def __init__(self):
-        self._storage: Dict[int, logging.Logger] = {}
-
-    def close(self):
-        for logger in self._storage.values():
-            _close_logger(logger)
-
-        self._storage = {}
-
-    def __getitem__(self, idx: int) -> logging.Logger:
-        """Get logger object"""
-        if idx not in self._storage:
-            self._storage[idx] = self._create_logger(idx)
-        return self._storage[idx]
-
-    def _create_logger(self, _: int) -> logging.Logger:
-        raise NotImplementedError
-
-
-class ProjectLoggerStorage(IndexedLogManager):
-    def _create_logger(self, pid):
-        project = _get_project(pid)
-
-        logger = logging.getLogger('cvat.server.project_{}'.format(pid))
-        server_file = logging.FileHandler(filename=project.get_log_path())
-        formatter = logging.Formatter(LOGGING['formatters']['standard']['format'])
-        server_file.setFormatter(formatter)
-        logger.addHandler(server_file)
-
-        return logger
-
-
-class TaskLoggerStorage(IndexedLogManager):
-    def _create_logger(self, tid):
-        task = _get_task(tid)
-
-        logger = logging.getLogger('cvat.server.task_{}'.format(tid))
-        server_file = logging.FileHandler(filename=task.get_log_path())
-        formatter = logging.Formatter(LOGGING['formatters']['standard']['format'])
-        server_file.setFormatter(formatter)
-        logger.addHandler(server_file)
-
-        return logger
-
-class JobLoggerStorage(IndexedLogManager):
-    def _create_logger(self, jid):
-        job = _get_job(jid)
-        return slogger.task[job.segment.task.id]
-
-class CloudSourceLoggerStorage(IndexedLogManager):
-    def _create_logger(self, sid):
-        cloud_storage = _get_storage(sid)
-
-        logger = logging.getLogger('cvat.server.cloud_storage_{}'.format(sid))
-        server_file = logging.FileHandler(filename=cloud_storage.get_log_path())
-        formatter = logging.Formatter(LOGGING['formatters']['standard']['format'])
-        server_file.setFormatter(formatter)
-        logger.addHandler(server_file)
-
-        return logger
-
-@define(slots=False)
-class _AggregateLogManager(LogManager):
-    def close(self):
-        for logger in vars(self).values(): # vars is incompatible with slots
-            if hasattr(logger, 'close'):
-                logger.close()
-
-@define(slots=False)
-class ServerLogManager(_AggregateLogManager):
-    project = field(factory=ProjectLoggerStorage)
-    task = field(factory=TaskLoggerStorage)
-    job = field(factory=JobLoggerStorage)
-    cloud_storage = field(factory=CloudSourceLoggerStorage)
-    glob = field(factory=lambda: logging.getLogger('cvat.server'))
-
-slogger = ServerLogManager()
```

Comment:
IMO, something like `slogger` would still be useful to ensure consistent formatting for common context data and to shorten repetitive code. I think something like this would be a good replacement:

```python
class _LoggerAdapterMapping:
    def __init__(self, logger: logging.Logger, object_type: str) -> None:
        self._logger = logger
        self._object_type = object_type

    def __getitem__(self, id_: int) -> logging.LoggerAdapter:
        return logging.LoggerAdapter(self._logger, {"cvat_context": f"{self._object_type}.id={id_}"})

class ServerLogManager:
    def __init__(self, logger_name: str) -> None:
        self.glob = logging.getLogger(logger_name)
        self.project = _LoggerAdapterMapping(self._logger, "Project")
        # ditto for task, job, etc...
```

In the individual files you would then just replace

```python
from cvat.apps.engine.log import slogger
```

with

```
from cvat.apps.engine.log import ServerLogManager

slogger = ServerLogManager(__name__)
```

and everything else would work as before.

You would also need a custom formatter class:

```python
class CvatContextFormatter(logging.Formatter):
    def format(self, record: logging.LogRecord) -> str:
        s = super().format(record)

        if context := getattr(record, "cvat_context", None):
            s = f'[{context}] {s}'
        
        return s
```

and to update the `LOGGING` configuration key to use it.

Reply:
I will close the PR and open the new one.